includes/class-wc-calypso-bridge-free-trial-payment-restrictions.php

```diff
@@ -0,0 +1,355 @@
+			return;
+		}
+
+		// Unset all payment gateways.
+		add_filter( 'woocommerce_available_payment_gateways', function ( $gateways ) {
+			return array();
+		}, PHP_INT_MAX );
+
+		// Change the "No available payment methods" message.
+		add_filter( 'woocommerce_no_available_payment_methods_message', function ( $message ) {
+
+			$message = __( 'This store is not ready to accept orders. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' );
+
+			return $message;
+
+		}, PHP_INT_MAX );
+
+		add_filter( 'gettext', function ( $translated_text, $text, $domain ) {
+
+			if ( $domain === 'woocommerce' ) {
+				switch ( $text ) {
+					// Since we're removing all payment gateways, we need to change the "No available payment methods" message.
+					case 'No payment method provided.' :
+						$translated_text = __( 'Your order could not be placed. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' );
+						break;
+				}
+			}
+
+			return $translated_text;
+
+		}, PHP_INT_MAX, 3 );
+
+
+		// TODO: How do we handle translations?
+		add_action('wp_head', function(){
+			?>
+			<script type="text/javascript">
+				function myPluginGettextFilter( translation, text, domain ) {
+					if ( text === 'There are no payment methods available. This may be an error on our side. Please contact us if you need any help placing your order.' ) {
+						return 'There are no payment methods available as this store is in trial mode.';
+					}
+					return translation;
+				}
+
+				// Adding the filter
+				wp.hooks.addFilter(
+					'i18n.gettext_woocommerce',
+					'my-plugin/override-add-to-reusable-blocks-label', // TODO: What plugin name should we use?
+					myPluginGettextFilter
+				);
+			</script>
+			<?php
+		});
+
+		// Prevent orders on shortcode checkout.
+		add_action( 'woocommerce_before_checkout_process', function () {
+			throw new Exception( __( 'Your order could not be placed. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' ) );
+		}, PHP_INT_MAX );
+
+		// Prevent orders on shortcode checkout - PayPal removes the checkout button and replaces it.
+		add_action( 'woocommerce_after_checkout_validation', function ( $data, $errors ) {
+			$errors->add(
+				409,
+				__( 'Your order could not be placed. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' )
+			);
+		}, PHP_INT_MAX, 2 );
+
+		// Prevent orders on block checkout.
+		add_action( 'woocommerce_store_api_checkout_order_processed', function () {
+			throw new Automattic\WooCommerce\StoreApi\Exceptions\RouteException(
+				409,
+				__( 'Your order could not be placed. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' )
+			);
+		}, PHP_INT_MAX );
+
+		// Display an info message on the checkout page.
+		add_filter( 'the_content', function ( $content ) {
+			if ( ! is_checkout() ) {
```

Comment:
Can we move this check outside the filter? 🤔

Reply:
We can't - I've already tried it; `is_checkout` always returns false when outside the `add_filter`